lib/matplotlib/pyplot.py

```diff
@@ -1311,19 +1311,19 @@ def subplot_tool(targetfig=None):
     """
     Launch a subplot tool window for a figure.
 
-    A :class:`matplotlib.widgets.SubplotTool` instance is returned.
+    A `matplotlib.widgets.SubplotTool` instance is returned.
     """
     if targetfig is None:
         targetfig = gcf()
-
-    with rc_context({'toolbar': 'None'}):  # No nav toolbar for the toolfig.
-        toolfig = figure(figsize=(6, 3))
-    toolfig.subplots_adjust(top=0.9)
-
-    if hasattr(targetfig.canvas, "manager"):  # Restore the current figure.
-        _pylab_helpers.Gcf.set_active(targetfig.canvas.manager)
-
-    return SubplotTool(targetfig, toolfig)
+    with rc_context({"toolbar": "none"}):  # No navbar for the toolfig.
+        # Use new_figure_manager() instead of figure() so that the figure
+        # doesn't get registered with pyplot.
+        manager = new_figure_manager(-1, (6, 3))
+    manager.set_window_title("Subplot configuration tool")
+    tool_fig = manager.canvas.figure
+    tool_fig.subplots_adjust(top=0.9)
+    manager.show()
```

Comment:
The work around there is to put the show on a 0s timer on the event loop.  That will make it show as soon as the event loop is allowed to run again (which is either on return of control to the user or when `plt.show()` is called).

Reply:
do you want to put in that patch? given that you seem to know what to do...